### PR TITLE
Getting started article

### PR DESCRIPTION
**Pull request by heckj**
After getting the walk through of Hummingbird at the SSWG meeting, I scanned through the docs. The first thing that struct me was although your presentation led brilliantly with "getting started" information, there wasn't anything equivalent in the documentation, and the tutorial was buried a bit down.

So I riffed on your opening slides, after going through the process a couple of times, I created a draft Getting Started article and tweaked some organization in your top level docs to present it a bit more "up front".

I tried to generally match the style and voice in the existing documentation, and added a bit of cleanup so that it was both consistent and not redundant. Very happy to change or tear up any of this, nothing sacred in what I've written.

### CODE REVIEW

File: Hummingbird.docc/index.md [hunk 1]
```diff
@@ -55,9 +51,14 @@ Hummingbird is designed to require the least number of dependencies possible, bu
...
-### Tutorials
+### Hummingbird Extensions
 
-- <doc:Todos>
+Hummingbird is designed to require the least number of dependencies possible, but this means many features are unavailable to the core libraries. Additional features are provided through extensions. The Hummingbird repository comes with additional modules
+
+- ``HummingbirdJobs``
+- ``HummingbirdTLS``
+- ``HummingbirdHTTP2``
+- ``HummingbirdTesting``
```
Comment (adam-fowler) [inline Hummingbird.docc/index.md:61]: Building documentation brings up an error for these
```
warning: Linking to 'HummingbirdTLS' from a Topics group in 'doc://com.opticalaberration.hummingbird/documentation/index' isn't allowed
```

**Review by adam-fowler — COMMENTED** [review]
This is great stuff Joseph. I was thinking about doing something like this. Just hadn't got around to it. I think the overview needs a little beefed out. Either each paragraph needs more content or maybe merged.

File: Hummingbird.docc/index.md [hunk 2]
```diff
@@ -27,16 +27,12 @@ let app = Application(
...
 
-- ``HummingbirdJobs``: framework for pushing work onto a queue to be processed outside of a request (possibly by another server instance)
-- ``HummingbirdTLS``: TLS support.
-- ``HummingbirdHTTP2``: Support for HTTP2 upgrades.
-- ``HummingbirdTesting``: helper functions to aid testing Hummingbird projects.
+### Tutorials
 
-## Topics
+- <doc:GettingStarted>
+- <doc:Todos>
```
Comment (adam-fowler) [inline Hummingbird.docc/index.md:35]: Can you do the same in Hummingbird.md. Bring the tutorial to the top and add the getting started link

File: Hummingbird.docc/index.md [hunk 3]
```diff
@@ -55,9 +51,14 @@ Hummingbird is designed to require the least number of dependencies possible, bu
 - <doc:Sessions>
 - <doc:OneTimePasswords>
 
-### Tutorials
+### Hummingbird Extensions
```
Comment (adam-fowler) [inline Hummingbird.docc/index.md:54]: I know you are just copying what I had before but I prefer the below
```suggestion
### Extending Hummingbird
```

File: Hummingbird.docc/index.md [hunk 4]
```diff
@@ -27,16 +27,12 @@ let app = Application(
...
-### Hummingbird Extensions
-
-Hummingbird is designed to require the least number of dependencies possible, but this means many features are unavailable to the core libraries. Additional features are provided through extensions. The Hummingbird repository comes with additional modules 
+## Topics
 
-- ``HummingbirdJobs``: framework for pushing work onto a queue to be processed outside of a request (possibly by another server instance)
-- ``HummingbirdTLS``: TLS support.
-- ``HummingbirdHTTP2``: Support for HTTP2 upgrades.
-- ``HummingbirdTesting``: helper functions to aid testing Hummingbird projects.
+### Tutorials
```
Comment (adam-fowler) [inline Hummingbird.docc/index.md:32]:
```suggestion
### Getting Started
```

File: Hummingbird.docc/index.md (hunk 1 above)
Comment (heckj) [reply Hummingbird.docc/index.md:61]: I'll replicate these updates into the Hummingbird directory, and see about resolving the error - I didn't notice it yesterday when I shifted that content down from the overview

**Comment by heckj** [comment]
I tweaked things up up a bit more. i re-organized, moving the error packages into "See Also", which apparently allows the links elsewhere. That moves it quite a ways down the page to be lodged with the other "external packages", so if that detail is critical and we need to put it up front, I can move it back to the Overview, which also allows these links. Or I can do so narratively if that makes more sense.

While I was reviewing the abstracts and overviews, I saw "lightweight, modern, flexible server framework" repeated a LOT, so I trimmed that down a bit to keep it foremost in the abstracts (the one-liner under the heading) where I think it'll serve the most value.

I also collapsed up the top paragraph in Getting Started. I'm still not entirely happy with it, but it's OK. I'm going to let it sit and bug me for the afternoon, see if I come up with better wording or better flow there.

If I've gone too far on anything, please say - I can back any of this down.

**Comment by heckj** [comment]
This looks like it'll llkely conflict with #36 just a smidge. I'm good to rebase if you want to merge that first, or the other way around.

**Comment by adam-fowler** [comment]
> This looks like it'll llkely conflict with #36 just a smidge. I'm good to rebase if you want to merge that first, or the other way around.

Maybe we should just keep this limited to the getting started article and do the re-org in another PR.

**Comment by heckj** [comment]
works for me, I’ll strip this down to just the article

**Review by adam-fowler — APPROVED** [review]
This looks good. I'll merge unless there is something else you'd like to add

**Review by Joannis — APPROVED** [review]

**Comment by heckj** [comment]
merge away!

**Pull request merged**